x-pack/plugins/ml/common/types/trained_models.ts

```diff
@@ -66,6 +66,7 @@ export type PutTrainedModelConfig = {
     model_aliases?: string[];
   } & Record<string, unknown>;
   tags?: string[];
+  model_type?: TrainedModelType;
```

Comment:
I had to add this optional property to the request body type to make it work (it's an optional property on the ES API). Not sure if this requires additional changes somewhere else in the app code.